bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestBasicOps.java

```diff
@@ -162,8 +167,25 @@ public void testPutAlmostTooBigValue() throws IOException {
   @Test
   /** Run a large value ten times for performance logging puproses */
   public void testPutAlmostTooBigValueTenTimes() throws IOException {
-    for (int i = 0; i < 10; i++) {
-      testPutGetDeleteExists(10 << 20, true, true); // 10 MB
+    ExecutorService es = Executors.newCachedThreadPool();
+    try {
+      for (int i = 0; i < 10; i++) {
+        es.submit(new Callable<Void>() {
+          public Void call() throws Exception {
+            testPutGetDeleteExists(10 << 20, true, true); // 10 MB
+            return null;
+          };
+        });
+      }
+    } finally {
+      es.shutdown();
+      while (!es.isTerminated()) {
+        try {
+          es.awaitTermination(1, TimeUnit.SECONDS);
```

Comment:
Why loop and await for 1 second at a time instead of a single awaitTermination()? Either way would be good to put an upper bound on amount of time to wait.